Frameworks/D2W/ERModernDirectToWeb/Sources/er/modern/directtoweb/components/buttons/ERMDRemoveRelatedButton.java

```diff
@@ -191,12 +194,14 @@ public Boolean showRemoveButton() {
     			EOEntity masterEntity = ERXEOAccessUtilities.entityForEo(masterObj);
     			EORelationship relationship = masterEntity.relationshipNamed(dds.detailKey());
     			EORelationship reverseRelationship = relationship.inverseRelationship();
-    			if(isRemoveable) {
-    				if(reverseRelationship == null) {
-    					_showRemoveButton = Boolean.TRUE;
-    				} else {
-    					_showRemoveButton = !reverseRelationship.isMandatory();
-    				}
+    			if(isRemoveable && !relationship.ownsDestination()) {
+                    if (!relationship.ownsDestination()) {
```

Comment:
What purpose has this check? This will always evaluate to true as in the line before it is already part of the if-condition.